dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/ds3panel/Ds3PanelPresenter.java

```diff
@@ -546,22 +548,22 @@ public void ds3DeleteObject(final boolean isButtonEvent) {
         if (Guard.isNullOrEmpty(values)) {
             if (root.getValue() == null) {
                 LOG.info("No files selected");
-                Ds3Alert.show(resourceBundle.getString("error"), resourceBundle.getString("noFiles"), Alert.AlertType.ERROR);
+                alert.showAlert(resourceBundle.getString("noFiles"));
             } else {
                 final ImmutableList.Builder<TreeItem<Ds3TreeTableValue>> builder = ImmutableList.builder();
-                values = builder.add(root).build().asList();
+                values = builder.add(root).build().asList();  //DM TODO why?
             }
         } else if (values.stream().map(TreeItem::getValue).anyMatch(value -> value.getType() == Ds3TreeTableValue.Type.Directory)) {
             if (isButtonEvent) {
```

Comment:
@rpmoore Is there any reason this should behave differently if its a ButtonEvent or from a right-click menu?  I can't think of any.  

The current behavior is that the Button event deletes the folder along with the item if its the only item in the folder.  The right-click menu correctly only deletes the item.